### PR TITLE
refactor: remove old names, use new solidity features, add file for poolClosing

### PR DESCRIPTION
**Pull request by offerijns**
- `distributor` => `reserve` rename (closes https://github.com/centrifuge/tinlake/issues/563)
- `epochTicker` => `coordinator` rename
- replace `ds-note` with custom events
- `uint(-1)` => `type(uint256).max`
- `hevm = Hevm(0x7109709ECfa91a80626fF3989D68f67F5b1DD12D);` => `hevm = Hevm(HEVM_ADDRESS);`
- add `file()` for `poolClosing` in coordinator

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,4 +37,4 @@ dapp test -m ':ContractName\.'
 For deploying the Tinlake contracts to mainnet or a testnet, view our deploy scripts in [Tinlake Deploy](https://github.com/centrifuge/tinlake-deploy).
 
 ## Community
-Join our public Slack channel: [Centrifuge Slack](http://centrifuge.io/slack).
+Join our public Discord: [Centrifuge Discord](https://centrifuge.io/discord/).
```
Comment (manuelpolzhofer) [inline README.md:40]: good catch

File: src/borrower/feed/navfeed.sol [hunk 2]
```diff
@@ -156,7 +155,7 @@ contract NAVFeed is BaseNFTFeed, Interest, Buckets, FixedPoint {
     // On borrow: the discounted future value of the asset is computed based on the loan amount and addeed to the bucket with the according maturity Date
     function _borrow(uint loan, uint amount) internal returns(uint navIncrease) {
         // ceiling check uses existing loan debt
-        require(ceiling(loan) >= safeAdd(borrowed[loan], amount), "borrow-amount-too-high");
+        require(ceiling(loan) >= amount, "borrow-amount-too-high");
```
Comment (manuelpolzhofer) [inline src/borrower/feed/navfeed.sol:158]: I am wondering if we actually need both methods:

```javascript
 function ceiling(uint loan) public view returns (uint) {
        if (borrowed[loan] > currentCeiling(loan)) {
            return 0;
        }
        return safeSub(currentCeiling(loan), borrowed[loan]);
    }

function currentCeiling(uint loan) public view returns(uint) {
        bytes32 nftID_ = nftID(loan);
        return rmul(nftValues[nftID_], ceilingRatio[risk[nftID_]]);
    }
```

We only use `ceiling` here it seems. We could just use `currentCeiling` and remove the `ceiling` function.

File: src/borrower/feed/test/navfeed.t.sol [hunk 3]
```diff
@@ -23,7 +23,7 @@ contract NAVTest is DSTest, Math {
     Hevm hevm;
 
     function setUp() public {
-        hevm = Hevm(0x7109709ECfa91a80626fF3989D68f67F5b1DD12D);
+        hevm = Hevm(HEVM_ADDRESS);
```
Comment (manuelpolzhofer) [inline src/borrower/feed/test/navfeed.t.sol:26]: This is nice

File: src/lender/definitions.sol [hunk 4]
```diff
@@ -28,7 +28,7 @@ contract Definitions is FixedPoint, Math {
             currSeniorAsset, newReserve, nav), nav, newReserve);
     }
 
-/// calculates the net wealth in the system
+    /// calculates the net wealth in the system
```
Comment (manuelpolzhofer) [inline src/lender/definitions.sol:31]: We can change the `///`  to `//`, we started this pattern but we are not consistent

File: src/test/system/test_suite.sol [hunk 5]
```diff
@@ -46,7 +46,7 @@ contract TestSuite is BaseSystemTest {
         // borrow loans with default maturity date 5 days from now
         uint maturityDate = DEFAULT_MATURITY_DATE;
         uint nftPrice = borrowAmount*3;
-        (uint loan, ) = setupOngoingLoan(nftPrice, borrowAmount, nftFeed.uniqueDayTimestamp(now) +maturityDate);
+        (uint loan, ) = setupOngoingLoan(nftPrice, borrowAmount, nftFeed.uniqueDayTimestamp(block.timestamp) +maturityDate);
```
Comment (manuelpolzhofer) [inline src/test/system/test_suite.sol:49]:
```suggestion
        (uint loan, ) = setupOngoingLoan(nftPrice, borrowAmount, nftFeed.uniqueDayTimestamp(block.timestamp) + maturityDate);
```

File: src/test/system/lender/lender_scenarios.t.sol [hunk 6]
```diff
@@ -427,16 +427,16 @@ contract LenderSystemTest is TestSuite, Interest {
...
-        hevm.warp(now + 1 days);
+        hevm.warp(block.timestamp + 1 days);
         coordinator.closeEpoch();
 
         // borrow loans maturity date 5 days from now
         uint borrowAmount = 100 ether;
         uint nftPrice = 200 ether;
         uint maturityDate = 5 days;
-        (uint loan, ) = setupOngoingLoan(nftPrice, borrowAmount, false, nftFeed.uniqueDayTimestamp(now) +maturityDate);
+        (uint loan, ) = setupOngoingLoan(nftPrice, borrowAmount, false, nftFeed.uniqueDayTimestamp(block.timestamp) +maturityDate);
```
Comment (manuelpolzhofer) [inline src/test/system/lender/lender_scenarios.t.sol:437]:
```suggestion
        (uint loan, ) = setupOngoingLoan(nftPrice, borrowAmount, false, nftFeed.uniqueDayTimestamp(block.timestamp) + maturityDate);
```

**Review by manuelpolzhofer — APPROVED** [review]
LGTM

File: src/lender/definitions.sol (hunk 4 above)
Comment (offerijns) [reply src/lender/definitions.sol:31]: Good point! Changed.

File: src/borrower/feed/navfeed.sol (hunk 2 above)
Comment (offerijns) [reply src/borrower/feed/navfeed.sol:158]: I think this requires a bit more investigation whether we're not relying on either externally. I'll add a separate ticket for this.

**Pull request merged**